### PR TITLE
fix(css): scope ProseMirror CSS to prevent bleeding into host apps (SD-1850)

### PR DESCRIPTION
**Pull request by tupizz**
## Problem

SuperDoc's CSS was leaking into host applications in three ways:

1. **Bare `a` element selector** — `a { text-decoration: auto }` in `global.css` overrode ALL `<a>` tags on the page, not just links inside the editor.

2. **Unscoped `.ProseMirror` selectors** — All ~55 `.ProseMirror` CSS rules in `prosemirror.css` affected any ProseMirror instance on the page. If a host app used ProseMirror for another editor (comments, chat), SuperDoc's styles would strip list styling, force table collapse, override image sizing, etc.

3. **`.sr-only` utility class collision** — The DomPainter injected a `.sr-only` class at runtime that collides with Bootstrap, Tailwind, and many other frameworks that use the same class name.

## Solution

This follows the **CKEditor 5 model** — the industry standard for editor CSS isolation. CKEditor scopes all content styles under `.ck-content`, we scope ours under `.super-editor`.

SuperDoc already did this for ~90% of its CSS. This PR closes the remaining gaps:

- **Remove bare `a` selector** from `global.css` — the scoped `.super-editor a` rule already handles editor links
- **Scope all `.ProseMirror` and `.ProseMirror-*` selectors** under `.super-editor` in `prosemirror.css` (~55 selectors)
- **Scope `.sd-*` selectors** in extension CSS files: `structured-content.css`, `document-section.css`, `noderesizer.css`, `ai.css`, `page-number.css`
- **Namespace keyframe animations**: `aiTextAppear` → `superdoc-aiTextAppear`, `ai-pulse` → `superdoc-ai-pulse`
- **Rename `.sr-only` → `.superdoc-sr-only`** in DomPainter runtime styles

### Why this approach

| Approach | Prevents outward bleed | Browser support | Works with `import 'lib/style.css'` |
|----------|:---------------------:|:--------------:|:-----------------------------------:|
| **Parent class scoping (this PR)** | Yes | 100% | Yes |
| Shadow DOM | Yes | ~96% | No (breaks ProseMirror) |
| CSS `@layer` | Yes | ~95% | Requires consumer changes |
| CSS Modules | Yes | 100% | No (build-time only) |

Parent class scoping is the simplest, most compatible approach. It's what CKEditor 5 and Quill use in production.

### What is NOT changed

- **`superdoc/` package CSS** — already scoped to `.superdoc`, `.super-editor`, `.superdoc-toolbar`
- **`isolation.css`** — inward protection (`all: revert`), already correctly scoped
- **DomPainter runtime styles** — already scoped to `.superdoc-*` (except `.sr-only` which we fix)
- **Tippy/popover styles** — render in `document.body`, kept global with unique `[data-theme]` selectors
- **`.presentation-editor__*` selectors** — already namespaced, render in DomPainter layer

## Test plan

- [x] `pnpm --filter super-editor test` — 672 files, 6345 tests pass
- [x] `pnpm --filter @superdoc/painter-dom test` — 23 files, 629 tests pass (`.superdoc-sr-only` assertions updated)
- [x] New `css-no-bleed.test.js` lint test — structurally verifies no bare element selectors, all `.ProseMirror` scoped under `.super-editor`, no generic utility class collisions (catches future regressions)
- [ ] Visual regression tests (CI)
- [ ] Manual verification: load editor in host app, confirm host styles unaffected

### CODE REVIEW

File: packages/super-editor/src/assets/styles/elements/prosemirror.css
Comment (caio-pizzol) [inline packages/super-editor/src/assets/styles/elements/prosemirror.css]: `@keyframes` names are global -- parent selectors don't scope them. if a host app also loads `prosemirror-gapcursor` CSS, both define ProseMirror-cursor-blink and last-loaded wins.

rename to `superdoc-cursor-blink` like you did for `superdoc-aiTextAppear` and `superdoc-ai-pulse`.

File: packages/super-editor/src/tests/css-no-bleed.test.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    if (entry.isDirectory()) {
+      results.push(...findCssFiles(join(dir, entry.name), rel));
+    } else if (entry.name.endsWith('.css')) {
+      results.push(rel);
+    }
+  }
+  return results;
+}
+
+/**
```
Comment (caio-pizzol) [inline packages/super-editor/src/tests/css-no-bleed.test.js:31]: postcss is already in the dep tree and can replace `extractTopLevelSelectors` in ~5 lines with proper edge case handling (unicode, @container, nested @supports).

the hand-rolled parser works for now but will miss things.

File: packages/super-editor/src/tests/css-no-bleed.test.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+ */
+const BARE_ELEMENT_PATTERN =
+  /^(a|p|div|span|table|tr|td|th|ul|ol|li|h[1-6]|img|pre|blockquote|code|input|button|select|textarea|form|label|section|article|nav|header|footer|main|aside)\s*(\{|,|:|\[|>|\+|~|$)/;
+
+/**
+ * Check if a selector is scoped under a SuperDoc parent class.
+ */
+function isScopedSelector(selector) {
+  // Already scoped under .super-editor, .superdoc, .sd-, .presentation-editor, .tippy-box
+  const scopedPrefixes = ['.super-editor', '.superdoc', '.sd-', '.presentation-editor', '.tippy-box', '.toolbar-icon'];
```
Comment (caio-pizzol) [inline packages/super-editor/src/tests/css-no-bleed.test.js]: `.toolbar-icon` is a generic name, not a superdoc namespace -- weaker than the other entries here.

could scope those ai.css toolbar selectors under `.super-editor` instead.

File: packages/super-editor/src/tests/css-no-bleed.test.js [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+  return scopedPrefixes.some((prefix) => selector.startsWith(prefix));
+}
+
```
Comment (caio-pizzol) [inline packages/super-editor/src/tests/css-no-bleed.test.js:113]: no check for `@keyframes` names -- that's how ProseMirror-cursor-blink slipped through.

a regex for `@keyframes` not prefixed with superdoc- or sd-would catch it.

**Review by harbournick — APPROVED** [review]
LGTM

**Comment by harbournick** [comment]
:tada: This PR is included in **superdoc** v1.17.0

The release is available on [GitHub release](https://github.com/superdoc-dev/superdoc/releases/tag/v1.17.0)
